src/fondant/component/component.py

```diff
@@ -76,6 +100,20 @@ def transform(self, dataframe: pd.DataFrame) -> pd.DataFrame:
             dataframe: A Pandas dataframe containing a partition of the data
         """
 
+    @classmethod
+    def resolve_produces(
+        cls,
+        consumes: t.Mapping[str, Field],
+    ) -> t.Mapping[str, t.Union[str, pa.DataType]]:
+        """Simulate a transform function and collect schema information."""
+        input_schema = cls.convert_consumes_to_dataframe_schema(consumes)
+
+        input_df = input_schema.example(size=5)
+        output_df = cls(consumes=consumes, produces={}).transform(dataframe=input_df)
+        output_schema = pandera.infer_schema(output_df)
```

Comment:
I guess the simulation is happening by passing a small subset of the data through the `transform` function. This can be problematic in two cases:

* How well is pandera able to simulate all the different transforms? Can we try this approach with some of our current existing components and see if it matches the expectations? 
* This would require all the dependencies required for the transform function to be installed during compile time which is simply not feasible 

Before going ahead with the implementation, might be best to check on both of those assumptions. Otherwise we might just need to opt for another approach

Reply:
I've added a try/catch block and a warning if module is not available. 
An user would have to provide a produces schema manually if the schema can't be detected automatically due to missing requirements.